src/facade/dragonfly_connection.cc

```diff
@@ -254,8 +254,9 @@ thread_local vector<Connection::PipelineMessagePtr> Connection::pipeline_req_poo
 thread_local Connection::QueueBackpressure Connection::tl_queue_backpressure_;
 
 void Connection::QueueBackpressure::EnsureBelowLimit() {
-  ec.await(
-      [this] { return subscriber_bytes.load(memory_order_relaxed) <= subscriber_thread_limit; });
+  ec.await([this] {
+    return done || subscriber_bytes.load(memory_order_relaxed) <= subscriber_thread_limit;
```

Comment:
And we had a deadlock if this was waiting while the connection fiber was joining

Reply:
the only call I find is here https://github.com/dragonflydb/dragonfly/blob/main/src/facade/dragonfly_connection.cc#L1720

where a connection A waits for the connection B to be below the limit.
can you describe the deadlock  what connection was joining during shutdown?

this fix is most probably incorrect because `done` is not atomic var and this is a multi-threaded function.